src/Admin/AbstractAdmin.php

```diff
@@ -1567,6 +1567,13 @@ public function getMaxPageLinks()
 
     public function getFormGroups()
     {
+        if (!\is_array($this->formGroups) && 'sonata_deprecation_mute' !== (\func_get_args()[0] ?? null)) {
```

Comment:
This is right, but how could someone fix this break?

Reply:
I think the simplest option is this:
https://github.com/sonata-project/SonataAdminBundle/blob/7abd1b4e345dcc540cf743bf03bdec744eaf9078/tests/Form/FormMapperTest.php#L66-L68

Other option could be to add some configuration option to tell our codebase it must behave according the relaxed and deprecated API or the new narrower one.

Finally, we could just omit these deprecations and expose these changes in the UPGRADE document.